src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationOptions.cs

```diff
@@ -27,41 +31,77 @@ public LinkedInAuthenticationOptions()
             AuthorizationEndpoint = LinkedInAuthenticationDefaults.AuthorizationEndpoint;
             TokenEndpoint = LinkedInAuthenticationDefaults.TokenEndpoint;
             UserInformationEndpoint = LinkedInAuthenticationDefaults.UserInformationEndpoint;
+            EmailAddressEndpoint = LinkedInAuthenticationDefaults.EmailAddressEndpoint;
+
+            Scope.Add("r_liteprofile");
+            Scope.Add("r_emailaddress");
 
             ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "id");
             ClaimActions.MapJsonKey(ClaimTypes.Email, "emailAddress");
-            ClaimActions.MapJsonKey(ClaimTypes.Name, "formattedName");
-            ClaimActions.MapJsonKey(ClaimTypes.GivenName, "firstName");
-            ClaimActions.MapJsonKey(ClaimTypes.Surname, "lastName");
-            ClaimActions.MapJsonKey(Claims.MaidenName, "maidenName");
-            ClaimActions.MapJsonKey(Claims.ProfileUrl, "publicProfileUrl");
-            ClaimActions.MapJsonKey(Claims.PictureUrl, "pictureUrl");
-            ClaimActions.MapJsonKey(Claims.Industry, "industry");
-            ClaimActions.MapJsonKey(Claims.Summary, "summary");
-            ClaimActions.MapJsonKey(Claims.Headline, "headline");
-            ClaimActions.MapCustomJson(Claims.Positions, user => user["positions"]?.ToString());
-            ClaimActions.MapJsonKey(Claims.PhoneticFirstName, "phoneticFirstName");
-            ClaimActions.MapJsonKey(Claims.PhoneticLastName, "phoneticLastName");
-            ClaimActions.MapJsonKey(Claims.FormattedPhoneticName, "formattedPhoneticName");
-            ClaimActions.MapCustomJson(Claims.Location, user => user["location"]?.ToString());
-            ClaimActions.MapJsonKey(Claims.Specialties, "specialties");
-            ClaimActions.MapJsonKey(Claims.NumConnections, "numConnections");
-            ClaimActions.MapJsonKey(Claims.NumConnectionsCapped, "numConnectionsCapped");
-            ClaimActions.MapJsonKey(Claims.CurrentShare, "currentShare");
-            ClaimActions.MapCustomJson(Claims.PictureUrls, user => user["pictureUrls"]?.ToString());
+            ClaimActions.MapCustomJson(ClaimTypes.Name, user => $"{GetMultiLocaleString(user, "firstName")} {GetMultiLocaleString(user, "lastName")}");
+            ClaimActions.MapCustomJson(ClaimTypes.GivenName, user => GetMultiLocaleString(user, "firstName"));
+            ClaimActions.MapCustomJson(ClaimTypes.Surname, user => GetMultiLocaleString(user, "lastName"));
+            ClaimActions.MapCustomJson(Claims.PictureUrl, user
+                => user.SelectTokens("$.profilePicture..elements[*].identifiers[0].identifier").Select(u => u.Value<string>()).LastOrDefault());
```

Comment:
Again regarding use of `SelectTokens(...)` here and on the line below.

Reply:
Same as above, I extracted this in a private static method using Linq syntax